orangecontrib/text/widgets/owtwitter.py

```diff
@@ -202,91 +278,95 @@ def mode_toggle(self):
+            self.start(
+                search,
+                self.api,
+                max_tweets,
+                self.word_list,
+                self.collecting,
+                self.language,
+                self.allow_retweets,
+                "content",
+            )
         else:
             if not self.word_list:
                 self.Error.empty_authors()
                 return None
-            if not any(a.startswith('@') for a in self.word_list):
+            if not any(a.startswith("@") for a in self.word_list):
                 self.Error.wrong_authors()
                 return None
-            return self.api.search_authors(max_tweets=max_tweets,
-                                           authors=self.word_list,
-                                           collecting=self.collecting)
+            self.start(
+                search,
+                self.api,
+                max_tweets,
+                self.word_list,
+                self.collecting,
+                None,
+                None,
+                "authors",
+            )
+        self.search_button.setText("Stop")
 
     def update_api(self, key):
         if key:
             self.Error.key_missing.clear()
-            self.api = twitter.TwitterAPI(key,
-                                          on_error=self.Error.api,
-                                          on_rate_limit=self.Error.rate_limit,
-                                          should_break=self.search.should_break,
-                                          on_progress=self.update_tweets_num)
+            self.api = twitter.TwitterAPI(key)
         else:
             self.api = None
 
-    @search.on_start
-    def on_start(self):
-        self.progressBarInit()
-        self.search_button.setText('Stop')
-        self.Outputs.corpus.send(None)
-        if self.mode == self.CONTENT and not self.limited_search:
-            self.progressBarFinished()
-
-    @search.on_result
-    def on_result(self, result):
-        self.search_button.setText('Search')
-        self.tweets_info_label.setText(self.tweets_info.format(len(result) if result else 0))
+    def on_done(self, result):
+        self.search_button.setText("Search")
+        if result:
+            self.info.set_output_summary(
+                len(result), f"{len(result)} tweets on output"
+            )
+        else:
+            self.info.set_output_summary(self.info.NoOutput)
         self.corpus = result
         self.set_text_features()
-        self.progressBarFinished()
 
-    @search.callback(should_raise=False)
-    def update_tweets_num(self, num=0, progress=None):
-        if self.limited_search or self.mode == self.AUTHOR:
-            if progress is not None:
-                self.progressBarSet(100 * progress)
-        self.tweets_info_label.setText(self.tweets_info.format(num))
+    def on_exception(self, ex):
+        if isinstance(ex, RateLimitError):
```

Comment:
Apparently 429 doesn't return an instance of RateLimitError, since I am getting a generic api error with 429.

Reply:
fixed